gobblin-config-management/src/main/java/gobblin/config/configstore/VersionComparator.java

```diff
@@ -0,0 +1,8 @@
+package gobblin.config.configstore;
+
+import java.util.Collection;
+
+public interface VersionComparator<V> {
```

Comment:
The SimpleVersionComparator/VersionComparator also check whether the String input is valid version or not. Hard to achieve that if only extends Comparator